### PR TITLE
 Added Content Length for empty body post / put requests.

### PR DESCRIPTION
**Pull request by prasa7**
## Purpose

Added Content Length for empty body post / put requests.

### CODE REVIEW

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java [hunk 1]
```diff
@@ -91,7 +92,8 @@
     private boolean keepAlive = true;
     /** logger for correlation.log */
     private static final Log correlationLog = LogFactory.getLog(PassThroughConstants.CORRELATION_LOGGER);
-
+    /** Chunked */
+    private String chunked = "chunked";
```
Comment (a5anka) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java]: Have we used the variable anywhere? Looks like an unused variable to me?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java [hunk 2]
```diff
@@ -230,6 +232,12 @@ public void start(NHttpClientConnection conn) throws IOException, HttpException
         } else {
             request = new BasicHttpRequest(method, path,
                     version != null ? version : HttpVersion.HTTP_1_1);
+            if (requestMsgCtx.getProperty(PassThroughConstants.DISABLE_CHUNKING_CHECK) != null &&
+                    requestMsgCtx.getProperty(PassThroughConstants.DISABLE_CHUNKING_CHECK).equals(false)) {
+                //TODO : Add "Transfer-Encoding: chunked" header
```
Comment (a5anka) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java]: Can we implement the TODO before merging?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/PassThroughHttpSender.java [hunk 3]
```diff
@@ -268,6 +268,8 @@ public InvocationResponse invoke(org.apache.axis2.context.MessageContext msgCont
                         && msgContext.getProperty(PassThroughConstants.ORGINAL_CONTEN_LENGTH) != null) {
                     msgContext.setProperty(PassThroughConstants.PASSTROUGH_MESSAGE_LENGTH, Long.parseLong(
                             (String) msgContext.getProperty(PassThroughConstants.ORGINAL_CONTEN_LENGTH)));
+                } else if (msgContext.getProperty(PassThroughConstants.DISABLE_CHUNKING) == null) {
+                    msgContext.setProperty(PassThroughConstants.DISABLE_CHUNKING_CHECK, false);
```
Comment (a5anka) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/PassThroughHttpSender.java]: It seems we always set this property to false. Do we need to introduce a new property for `DISABLE_CHUNKING_CHECK`?

**Comment by indikasampath2000** [comment]
Closing the PR because https://github.com/wso2/wso2-synapse/pull/1271 sent to the master branch fixing the issue.

**Pull request closed without merging**